src/youtube/js/youtube.js

```diff
@@ -0,0 +1,374 @@
+   * @property {object} encodings An array of supported encoding types (ex. OO.VIDEO.ENCODING.MP4)
+   * @property {object} features An array of supported features (ex. OO.VIDEO.FEATURE.BITRATE_CONTROL)
+   * @property {string} technology The core video technology (ex. OO.VIDEO.TECHNOLOGY.HTML5)
+   */
+  var OoyalaYoutubeVideoFactory = function() {
+    this.name = pluginName;
+    this.features = [ OO.VIDEO.FEATURE.VIDEO_OBJECT_SHARING_GIVE,
+                      OO.VIDEO.FEATURE.BITRATE_CONTROL ];
+    this.technology = OO.VIDEO.TECHNOLOGY.HTML5;
+    this.encodings = [OO.VIDEO.ENCODING.MP4];
+
+    /**
+     * Creates a video player instance using OoyalaYoutubeVideoWrapper.
+     * @public
+     * @method OoyalaYoutubeVideoFactory#create
+     * @param {object} parentContainer The jquery div that should act as the parent for the video element
+     * @param {string} domId The id of the video player instance to create
+     * @param {object} controller A reference to the video controller in the Ooyala player
+     * @param {object} css The css to apply to the video element
+     * @param {string} playerId An id that represents the player instance
+     * @returns {object} A reference to the wrapper for the newly created element
+     */
+    this.create = function(parentContainer, domId, controller, css, playerId) {
+      youtubeVideoContainer = parentContainer;
+      player= document.createElement('div');
+      player.id ="player";
+      player.style.position = "absolute";
+      player.style.top = "0px";
+      player.style.left = "0px";
+
+      $('head').append("<script type = 'text/javascript' src = 'http://www.youtube.com/iframe_api'></script>");
+      window.onYouTubePlayerAPIReady = function() { onYouTubeIframeAPIReady(); }; 
+      element = new OoyalaYoutubeVideoWrapper();
+      element.controller = controller;
+      controller.notify(controller.EVENTS.CAN_PLAY);
+      youtubeVideoContainer.append(player);
+      return element;
+    };
+
+    /**
+     * Destroys the video technology factory
+     * @public
+     * @method OoyalaVideoFactory#destroy
+     */
+    this.destroy = function() {
+      this.encodings = [];
+      this.create = function() {};
+    };
+
+    this.maxSupportedElements = -1;
+  };
+
+  /* 
+   *This function creates an <iframe> (YouTube player)
+   *after the API code downloads.
+   */
+  function onYouTubeIframeAPIReady() {
+    youtubePlayer = new YT.Player('player', {
+      videoId: youtubeID,
+      height: "100%",
+      width: "100%",
+      playerVars: { 'autoplay': 0, 'controls': 0 },
+      events: {
+      'onReady': onPlayerReady,
+      'onPlaybackQualityChange': onPlayerPlaybackQualityChange,
+      'onStateChange': onPlayerStateChange,
+      'onError': onPlayerError
+      }
+    });
+  };
+
+  /*
+   *The Youtube iframe API will call this function when the video player is ready.
+   */
+  function onPlayerReady(event) {
+    playerReady = true;
+    if (!playerNotready) {
```

Comment:
How do playerReady and playerNotready interact with each other? From an outside point of view, there should only be one boolean. Can you explain the purpose of these two flags?